CIP-XXXX/README.md

```diff
@@ -0,0 +1,116 @@
+---
+CIP: ????
+Title: Plutus support for Multi-Scalar Multiplication over BLS12-381
+Status: Proposed
+Category: Plutus
+Authors:
+    - Dmytro Kaidalov <dmytro.kaidalov@iohk.io>
+    - Adam Smolarek <adam.smolarek@iohk.io>
+    - Thomas Vellekoop <thomas.vellekoop@iohk.io>
+Implementors:
+    - IOG Plutus team
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/884
+Created: 2024-08-22
+License: CC-BY-4.0
+---
+## Abstract
+The CIP proposes an extension of the current Plutus functions to provide support for the efficient computation of the multi-scalar multiplication over the BLS12-381 curve. This operation is crucial in a number of cryptographic protocols that can enhance the capabilities of the Cardano blockchain.
+
+## Motivation: why is this CIP necessary?
+Multi-scalar multiplication (MSM) is an algebraic group operation of the following form. Let $G$ be a group of prime order $p$. Let $g_0, g_1, ..., g_{N-1}$ be elements of $G$ and let $e_0, e_1, ..., e_{N-1}$ be elements of $Z_p$. Then, the multi-scalar multiplication $M$ is calculated as $M=\sum_{i=0}^{N-1} e_i \cdot g_i$.
+
+This operation appears in many cryptographic protocols. Its naive implementation requires $N$ scalar multiplications and $N$ group additions. However, the performance can be significantly improved by employing advanced algorithms, such as the [Pippenger Approach](https://hackmd.io/@tazAymRSQCGXTUKkbh1BAg/Sk27liTW9). Moreover, it can be further optimized for a particular group type (e.g., for elliptic curve groups).
```

Comment:
@dkaidalov : does it make sense to include some more information on these further optimizations - e.g., having some preliminary estimates for the improvements. For example, https://eprint.iacr.org/2022/1400.pdf proposes two improvements for twists of Edward curves (so should be applicable to BLS curves).

Reply:
thanks, @djetchev 
we definitely can add some links for those interested in details. 
Although, I don't think we can consider any further optimizations except those that are already implemented in the blst library.
With this CIP the idea is just to extend the Plutus API with the function from blst that implements MSM. The blst library is widely adopted, tested, formally verified and is already a dependency for Cardano, so it should be relatively straightforward.
Considering some additional optimizations means that we need to develop new cryptography, which complicate things in orders of magnitude.